### PR TITLE
add two heterodyne / relative likelihood models dedicated to sky marginalization

### PR DESCRIPTION
**Pull request by ahnitz**
This adds two new models of the heterodyne type. One is focused on 22-mode only optimizations, while the other is left more generic. An example is added for the latter (which inherits from the former). 

Depends on #4202

### CODE REVIEW

**Comment by spxiwh** [comment]
@ahnitz I think this depends on some ohter patches that are open, so I'll try to review this promptly once the dependencies are all in! (Although others may be better reviewers than me on this!)

**Comment by ahnitz** [comment]
@cdcapano @spxiwh I think this should be ready to look at now.

**Comment by ahnitz** [comment]
@spxiwh @cdcapano Poke?

**Review by spxiwh — COMMENTED** [review]
This looks good to me, but others know this code better than me.

I think it would be good to add more explanation of what the new models do, and what the options in the `model` section of the ini file actually mean. Maybe it would be good to add details into the documentation to describe this??

**Comment by ahnitz** [comment]
@spxiwh All the options were part of previous PRs and not completely specific to these models. I agree having some more documentation on what the different models are / can do would be good. I've been thinking about putting together a table or something like that for the docs, might as well do that now, but in a separate PR.

File: pycbc/inference/models/relbin.py [hunk 1]
```diff
@@ -589,3 +607,184 @@ def extra_args_from_config(cp, section, skip_args=None, dtypes=None):
...
+                                   sdat['b0'], sdat['b1'])
+            filt += filter_i
+            norm += norm_i
+        loglr = self.marginalize_loglr(filt, norm)
+        return loglr
+
+
+class RelativeTimeDom(RelativeTime):
+    """ Heterodyne likelihood optimized for time marginalization
+    """
```
Comment (cdcapano) [inline pycbc/inference/models/relbin.py:708]: This needs some more documentation. What's the difference between this and `RelativeTime`? Also not crazy about the name... if `Domain` meant by `Dom`, just call it `RelativeTimeDomain`.

**Review by cdcapano — CHANGES_REQUESTED** [review]
Hard to check the math, but if you ran the tests in the examples and they look ok, it seems fine. Needs more documentation though (see comment). Also, I noticed you're running cprofile in the example script. Is that on purpose, or was that an accidental copy? If on purpose, add a comment explaining that so people don't think it's necessary to do that in order to run pycbc inference.

**Comment by cdcapano** [comment]
@ahnitz The more extensive docs you suggest in your reply to Ian sounds good, but I would still request at least a brief synopsis in the class doc string for `RelativeTime` and `RelativeTimeDom` for people looking at source code and/or using the help in Jupyter. Right now they're identical, which is confusing.

**Comment by ahnitz** [comment]
@spxiwh @cdcapano Sorry, it took me a while to get back to this. I've updated the docstrings to clarify the difference between them i.e. the RelativeTimeDom only works for Dominant-mode only waveforms.

**Review by cdcapano — APPROVED** [review]
Looks good.

**Pull request merged**